esy/Task.ml

```diff
@@ -8,12 +8,34 @@
 
 module ConfigPath = Config.ConfigPath
 
+let renderCommandExpr ?name ~system ~scope expr =
+  let pathSep =
+    match System.host with
+    | System.Darwin
+    | System.Linux
+    | System.Cygwin -> "/"
+    | System.Other _ -> "\\"
+  in
+  let colon =
+    match name, system with
+    | Some "OCAMLPATH", (System.Linux | System.Darwin) -> ":"
+    | Some "OCAMLPATH", (System.Cygwin | System.Other _) -> ";"
+    | _, (System.Linux | System.Darwin | System.Cygwin) -> ":"
+    | _, System.Other _ -> ";"
```

Comment:
The rule is "`:` on unix-y systems, `:` — on others"

Reply:
I believe we'll actually want `System.Cygwin` to behave the same as `System.Linux` and `System.Darwin`, and `System.Windows` to have the `;`

Reply:
Please see the updated diff. Yes, we `System.Cygwin` behaves as `System.Linux` and `System.Darwin` for all but `OCAMLPATH` which is a special case (documented by @jordwalke).